Converted old documentation to markdown

EkiH opened the pull request.

EkiH committed:
Converted old documentation to markdown

jvuoti commented on README.md:
```diff
@@ -1,2 +1,503 @@
+Path of the entity (queue, topic or subscription) to send or receive messages to or from. When sending messages, give the name of the queue or topic. When receiving messages, give the queue name, or if the entity is a subscription, the entire path to it, in the format [topic name]/subscriptions/[subscription name]
+    Things to note:
+    The topics, queues or subscriptions will not be created automatically. If they do not exist, an error will be thrown and the transfer will fail.
+    For source endpoints, only a single message will be received for a transfer, i.e. batch receive is not supported. The message will also be completed immediately after receive, even before it is written to disk to a temporary file. Therefore if you need guarantees that the messages are processed, you will need to create a custom endpoint that e.g. persists the messages being processed, or othe ...
```
Talking about custom endpoints?

jvuoti reviewed (commented):
Looks good. One mention about a custom endpoint though

jvuoti committed:
Update README.md

Removed reference to custom endpoints. Upped Cobalt parameters heading levels a bit so they look nicer

jvuoti commented:
Removed the ref about custom endpoints, tweaked the Source, Destination etc headers a bit

EkiH committed:
Added escape characters and changed formatting to fit GitHub's markdown

jvuoti commented on README.md:
```diff
@@ -9,10 +9,10 @@
+        - [Source](#source)
+        - [Destination](#destination)
+        - [Message processing steps](#message-processing-steps)
+        - [Parameters](#parameters)
```
Ah, are these now moved too high in the TOC as well? Should we just update the structure a bit, e.g. remove the "Cobalt task transfer parameters" level altogheter?

EkiH committed:
Simplified header structure
